server/status_test.go

```diff
@@ -118,9 +130,96 @@ func TestStatusJson(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
+			t.Log(string(body))
 			if matches, err := regexp.Match(spec.expected, body); !matches || err != nil {
 				t.Errorf("expected match: %t; err nil: %s", matches, err)
 			}
 		}
 	}
 }
+
+// TestStatusGossipJson ensures that the output requrest for the full gossip
+// info contains the required fields.
+func TestStatusGossipJson(t *testing.T) {
+	s := StartTestServer(t)
+	defer s.Stop()
+
+	type prefixedInfo struct {
+		Key string                 `json:"Key"`
+		Val []storage.PrefixConfig `json:"Val"`
+	}
+
+	type rangeDescriptorInfo struct {
+		Key string                `json:"Key"`
+		Val proto.RangeDescriptor `json:"Val"`
+	}
+
+	type keyValueStringPair struct {
+		Key string `json:"Key"`
+		Val string `json:"Val"`
+	}
+
+	type keyValuePair struct {
+		Key string                 `json:"Key"`
+		Val map[string]interface{} `json:"Val"`
+	}
+
+	type infos struct {
```

Comment:
One much simpler approach for these tests is to let JSON unmarshal the response directly into a go dict and then have a list of keys to check are in the dict.

    var f interface{}
    err := json.Unmarshal(b, &f)

Will unmarshal the byte slice `b` into `f`, which will be of type `map[string] interface{}`

Then your test could just loop through a slice of string keys and verify existing of each.